documentation/src/main/asciidoc/manual-index.asciidoc

```diff
@@ -361,3 +361,226 @@ parameters through - see <<lucene-indexing-performance>> for more details.
+====
+This feature is a work in progress.
+====
+
+The integration with JSR-352, Batch Applications for the Java Platform, is in development. We do
+think we have the basics covered and we are looking for feedback.
+
+Hibernate Search provides a new mass indexing approach using batch job introduced by JSR352.
+It covers not only the existing functionality of the previous indexer, but also benefits the
+powerful standard based facilities of the Java Batch Platform (JSR-352), such as checkpoint and
+step partitioning. However, it requires a batch runtime for the execution:
+
+* For Java EE users, we use the batch runtime provided by your Java EE container.
+* For Java SE users, we use the batch runtime
+  https://github.com/jberet/jsr352/tree/master/jberet-se[JBeret SE].
+
+.How to use the new mass-indexer?
+====
+[source, JAVA]
+----
+Properties jobProps = MassIndexingJob
+        .parameters()
+        .forEntity( MyEntity.class )
+        .build();
+long executionId = BatchRuntime
+        .getJobOperator()
+        .start( MassIndexingJob.NAME, jobProps );
+----
+====
+
+In the following paragraphs, we will explain why JSR-352 standard is a good choice, and how to
+configure different parameters in order to bootstrap the mass-indexing job.
+
+==== Define your indexing scope
+Our job allows you to define your own indexing scope—you can start a full indexation or a partial
+indexation through 3 different methods: selecting the desired entity types, using HQL, or using
+Hibernate criteria.
+
+.How to add define the indexing scope in different methods?
+====
+[source, JAVA]
+----
+// full indexation
+Properties jobProps1 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps2 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( "select c from MyClass c where c.name in ( 'Foo', 'Bar' )" )
+        .build();
+
+// partial indexation using HQL
+Properties jobProps3 = MassIndexingJob
+        .parameters()
+        .forEntity( MyClass.class )
+        .restrictedBy( Restrictions.in( "name", "Foo", "Bar" ) )
+        .build();
+----
+====
+
+==== Parallelism
+In order to maximize the performance, we highly recommend you to speed up the mass indexer using
+parallelism. Parallelism is activated by default. Under the standard JSR 352, the exact word is
+_"partitioning"_: The indexing step may run as multiple partitions, one per thread. Each partition
+has its own partition ID and parameters. If there’re more partitions than threads, partitions are\
+considered as a queue to consume: each thread can only run one partition at a time and won’t consume
+the next partition until the previous one is finished.
+
+===== Threads
+The maximum number of threads used by the job execution is defined through method `maxThreads()`.
+Within the N threads given, there’s 1 thread reserved for the core, so only N - 1 threads are
```

Comment:
> Is this always true, regardless of the JSR-352 implementation?

I didn't test it on other implementations except JBeret. According to spec 1.0, 11.7 Partitioned Chunk Processing (page 107), it seems that it is the case: there's one main thread `A` and x sub-threads `Px` for different partitions.

> Does it mean that setting maxThreads to 1 will not work?

It will work. In this case, we disable the multi-threading mechanism. This single thread will be used in both job operations and job executions. I updated the doc.